spring-batch-infrastructure/src/main/java/org/springframework/batch/item/file/builder/FlatFileItemWriterBuilder.java

```diff
@@ -459,9 +458,7 @@ protected DelimitedBuilder(FlatFileItemWriterBuilder<T> parent) {
 					"A list of field names or a field extractor is required");
 
 			DelimitedLineAggregator<T> delimitedLineAggregator = new DelimitedLineAggregator<>();
-			if (StringUtils.hasLength(this.delimiter)) {
-				delimitedLineAggregator.setDelimiter(this.delimiter);
-			}
+			delimitedLineAggregator.setDelimiter(this.delimiter);
```

Comment:
We should at least check that `this.delimiter` is not `null` (which was part of `StringUtils.hasLength`). Otherwise the new test `testDelimitedOutputWithEmptyDelimiter ` fails with a NPE in `StringJoiner.java:122` when I pass `null`.